esm/encode.js

```diff
@@ -11,8 +13,9 @@ export const encode = async (buffer, format = 'deflate') => {
     response = response.stream().pipeThrough(stream);
   }
   const view = new Uint8Array(await new Response(response).arrayBuffer());
-  const out = [];
-  for (let i = 0; i < view.length; i += 2000)
-    out.push(String.fromCharCode(...view.subarray(i, i + 2000)));
-  return btoa(out.join(''));
+  let out = '';
+  const viewLength = view.length;
```

Comment:
Same as with the other PR, immutable arrays have static length. Accessing it once or twice won’t save the world but its compression performance is worse